tmt/steps/discover/fmf.py

```diff
@@ -333,6 +333,41 @@ def assert_git_url(plan_name: Optional[str] = None) -> None:
                 if not self.opt('dry'):
                     shutil.copytree(directory, self.testdir, symlinks=True)
 
+        # Check for a presense of a file with rules for ref
+        # either following special syntax ref: @filepath
+        if ref and str(ref)[0:1] == '@':
+            ref_filepath = os.path.join(self.testdir, ref[1:])
+        # or using the default location
+        else:
+            ref_filepath = os.path.join(self.testdir, ".tmt/ref.fmf")
+        # dynamic ref check is enabled when ref is not defined or when it starts with @
+        if (not ref or str(ref)[0:1] == '@'):
+            if os.path.exists(ref_filepath):
+                self.debug(f"Dynamic 'ref' definition file '{ref_filepath}' exists")
+                # there is a file with rules
+                # read it, process it and get the value of the variable REF
+                try:
+                    with open(ref_filepath, encoding='utf-8') as datafile:
+                        data = tmt.utils.yaml_to_dict(datafile.read())
+                except OSError as error:
+                    raise tmt.utils.FileError(f"Failed to read '{ref_filepath}'.\n{error}")
+                # we need to built a dummy tree and a plan so we can evaluate fmf data
```

Comment:
TBH I have no idea if this is the right approach. I have somehow built it from pieces of code found here and there.

Reply:
I've used this instead and it seems to work. https://github.com/lukaszachy/tmt/commit/0b92b4dbc2aaeb8180ac053cbf69635f6193f633

Reply:
should be fixed now.

Reply:
@lukaszachy I am thinking.. Without `_expand_node_data`, will variable or context parametrization work?

Reply:
@lukaszachy FYI https://github.com/teemtee/tmt/pull/1607